soc/xtensa/intel_adsp/common/ipc.c

```diff
@@ -163,6 +162,7 @@ int intel_adsp_ipc_send_message(const struct device *dev,
 	config->regs->idd = ext_data;
 	config->regs->idr = data | INTEL_ADSP_IPC_BUSY;
 	k_spin_unlock(&devdata->lock, key);
+	pm_device_busy_clear(dev);
```

Comment:
Should we clear the busy bit also for -EBUSY on L157 ? The old implementation is a bit suspect as well, but I think the expected flow is that the ack (in irq) will clear the bit in case send returns -EBUSY. If we now change the busy flag clearing to happen here synchronously, also the EBUSY handling needs to be changed.